### ISSUE TITLE
Add data_source for Active Directory

### ISSUE THREAD

**Issue by okrause** [NetApp/terraform-provider-netapp-gcp#15]
## Use case
To be able to successfully create a SMB volume, an Active Directory configuration needs to be present for the region of interest. No AD, no SMB volume.

Therefore it makes sense to use the *depends_on" meta-argument of Terraform to establish a dependency of the SMB volume to an AD configuration.

Currently, this can be done if the AD configuration is also create and managed by TF and the modules have access to this TF configuration.

The more common case is, that the AD configuration was created manually or via another TF module which cannot easily be referenced.

Having an Active Directory data_source which can be used to query the existing AD configuration for the intended region would allow to make this AD configuration "known" to TF and would create an resource which the SMB volume can depend on.

Example code:
```tf
# Specify the provider and access details
locals {
  gcp_project = "1234567890"
  gcp_service_account = "/Users/test/cvs-admin-sa.json"
  network = "default"
  volume_name = "tftest"
  region = "europe-west3"
  size = 1024
  service_level = "standard"
}

terraform {
  required_version = ">= 0.13"
  required_providers {
    netapp-gcp = {
      source = "NetApp/netapp-gcp"
    }
  }
}

provider "netapp-gcp" {
  project         = local.gcp_project
  service_account = local.gcp_service_account
}

data "netapp-gcp_active_directory" "ad-west3" {
    # domain = "example.com"
    region = local.region
}

resource "netapp-gcp_volume" "gcp-smb-volume" {
  provider = netapp-gcp
  name = local.volume_name
  region = local.region
  protocol_types = ["SMB"]
  network = local.network
  size = local.size
  service_level = local.service_level
  depends_on = [
    data.netapp-gcp_active_directory.ad-west3
  ]
}
```

**Comment by wenjun666** [comment]
yes, we will add data_source for Active Directory.

**Comment by okrause** [comment]
5f2a5962509185660688fa8390061cd2ac197020 and a1914cb324c33a7f8159b1868255bacefafc6f83 added AD data source. I verified it for some use cases. Looks good. Thank you!

I modified my example files to make use of it. We need to talk when to include it. Now, or when the new functionality is available in the provider at the hashicorp repo. But this is not a discussion for this issue, this is done. Closing this one.
